### PR TITLE
Update nicematrix.cwl

### PR DESCRIPTION
**Pull request by Basmah1961**
The suggested changes allow for additional syntax auto-completion in the form `\begin{NiceArray}{*{5}{c}}[hvlines]` (i.e., options could be added after the preamble of `NiceTabular` environment). This is the typical format provided by the author of `NiceMatrix`

### CODE REVIEW

File: completion/nicematrix.cwl [hunk 1]
```diff
@@ -277,8 +286,8 @@ rules/width=##L
 rules/color=#%color
 #endkeyvals
 
-\SubMatrix{delim1}{i-j}{k-l}{delim2}#/NiceTabular,NiceTabular*,NiceTabularX,NiceArray,pNiceArray,bNiceArray,BNiceArray,vNiceArray,VNiceArray,NiceMatrix,pNiceMatrix,bNiceMatrix,BNiceMatrix,vNiceMatrix,VNiceMatrix,NiceArrayWithDelims
-\SubMatrix{delim1}{i-j}{k-l}{delim2}[options%keyvals]#/NiceTabular,NiceTabular*,NiceTabularX,NiceArray,pNiceArray,bNiceArray,BNiceArray,vNiceArray,VNiceArray,NiceMatrix,pNiceMatrix,bNiceMatrix,BNiceMatrix,vNiceMatrix,VNiceMatrix,NiceArrayWithDelims
+\SubMatrix{delim1%delim}{i-j}{k-l}{delim2%delim}#/NiceTabular,NiceTabular*,NiceTabularX,NiceArray,pNiceArray,bNiceArray,BNiceArray,vNiceArray,VNiceArray,NiceMatrix,pNiceMatrix,bNiceMatrix,BNiceMatrix,vNiceMatrix,VNiceMatrix,NiceArrayWithDelims
+\SubMatrix{delim1%delim}{i-j}{k-l}{delim2%delim}[options%keyvals]#/NiceTabular,NiceTabular*,NiceTabularX,NiceArray,pNiceArray,bNiceArray,BNiceArray,vNiceArray,VNiceArray,NiceMatrix,pNiceMatrix,bNiceMatrix,BNiceMatrix,vNiceMatrix,VNiceMatrix,NiceArrayWithDelims
```
Comment (muzimuzhi) [inline completion/nicematrix.cwl:290]: Currently it seems texstudio doesn't support multiple keyvals fields, either with the same or different keyvals list, in a single command.

Also the format should be
```
\SubMatrix{delim1%keyvals}...

#keyvals:\SubMatrix
...
#endkeyvals
```
This syntax also shows there's no way to specify different keyvals list for different arguments in a single command.

File: completion/nicematrix.cwl [hunk 2]
```diff
@@ -295,6 +304,16 @@ hvlines
 name=
 #endkeyvals
 
+#delim:\SubMatrix
```
Comment (sunderme) [inline completion/nicematrix.cwl:307]: #delim does not exist as command, keyvals ?

**Comment by sunderme** [comment]
there seems to be no feedback on the review ?
Do I see this PR as abandoned and can close it ?

**Comment by sunderme** [comment]
please come back with the first part (new commands).
The keyval definition is faulty.

**Pull request closed without merging**